### ISSUE TITLE
Implement preliminary template comments

### ISSUE THREAD

**Issue by lboxell** [gslab-econ/template#20]
First follow up to #19. The below points are a combination of comments from different people (many from MG's https://github.com/gslab-econ/template/issues/19#issuecomment-259769756).

Steps:
 - Combine large and small templates. We should only have one template in `master`. This should contain the superset of directories in small and large templates.

### Readme / documentation
* Lab-specific protocol info should be moved to RA manual
* Google drive should be dropped as a requirement; assume generic storage 
* Discussion of requirements could be pared down / simplified further
* Specify scons 2.4 or greater as (suggested?) pre-requisite because earlier versions use a different cache naming scheme (single alphanumeric for directories versus two alphanumerics for directories). 

### Setup
* I had to explicitly install the Python `requests` library to run the large template. Looks like this is something gslab_scons depends on. Is there a way to specify things so this (and any other dependencies) get installed automatically when we install gslab_python?
* I would like to define an explicit `setup.py` (or equivalent step within scons) which verifies that all prerequisites are satisfied. The goal should be that we would trap most errors at this stage so if a new user passes the setup step the repository runs without error with very high probability. It could also be part of this step to check out externals from SVN + create sim links to external raw data / code / etc.

### Other
* The SConscript in /data/ in the lg template should use build_r not env.Command()
* What about renaming the builders we define in SConstruct `BuildLyx`, `BuildStata`, etc. rather than `Lyx`, `Stata`, etc.?
* We should not use `from xx import *`; switch to `import xx` or `from xx import specific-commands`
* We should think about the templates we want R and Stata scripts to follow and implement those here. For Stata, this will mean revisiting our gslab-specific tools like `preliminaries` -- whether we want to keep using them and also how we want to install them.
* I don't think we want very short scripts like the current build.R to use the
```
main <- function() {
}
main()
```
structure. 

 - Add `ondeck.lyx` and `text.lyx` to /paper directory.
 - Create ./tables and ./figures subdirectories that live are at the same level as ./paper. These subdirectories should store a .lyx figure/table file for each figure or table that gets included in the paper.
 - Implement grammar suggestions from https://github.com/gslab-econ/template/issues/19#issuecomment-259566137

**Comment by lboxell** [comment]
Do all work in a branch and then we'll pull everything into `master` per standard procedures.

**Comment by lboxell** [comment]
Note that I added:
```
* Specify scons 2.4 or greater as (suggested?) pre-requisite because earlier versions use a different cache naming scheme (single alphanumeric for directories versus two alphanumerics for directories). 
```
to the instructions.

**Comment by yuchuan2016** [comment]
Hi @lboxell ,

I think it needs discussion about the templates that Stata and R scripts will follow. Should we still checkout the Statatools from svn and store them in /externals/?

**Comment by lboxell** [comment]
@yuchuan2016,

Per our discussion, I think you can either exclude these statatools from the template or use the git version if @gentzkow thinks we should keep them around (in which case, I think we should import them into `build/external`)

**Comment by gentzkow** [comment]
I think we should discuss what to do with Stata tools. I would ideally like
to set these up so they can be easily installed in Stata, just as we
install our libraries for Python.

**Comment by yuchuan2016** [comment]
Update:

Things that have been implemented:
- Large and small templates are combined.
- Lab-specific protocol has been deleted.
- Google drive is dropped as a requirement.
- SCons 2.4 or later and gslab_tools 1.1.1 or later are specified as pre-requisites.
- Builders defined in SConstruct are renamed as BuildLyx, BuildStata, etc.
- The SConscripts use BuildR, BuildLyx, etc.
- The structure for simple code in build.R is dropped.
- Ondeck.lyx and text.lyx to are added to /paper directory.
- Two subdirectories ./tables and ./figures are added.
- Grammar in README is modified.
- Instruction for how to use the template in Wiki is moved to README.

Things that have not been implemented / need discussion:
-  The templates for R and Stata scripts. Do we need to revisit gslab-specific tools like preliminaries and if so, how to install?
- An explicit setup.py (or equivalent step within icons) that verifies that all prerequisites are satisfied.
- Include `install_requires` in `gslab_python` that install all dependencies before installing `gslab_tools`.
- In SConsctruct in /paper or /talk, should we specific `figure.lyx` or `figure.eps` as source? I think the idea for having a /figures folder is to specify all `figure.lyx` as sources, but if so, changing the `figure.eps` will not automatically rebuild the paper.

**Comment by lboxell** [comment]
@yuchuan2016,

With regards to your last bullet point, I think once https://github.com/gslab-econ/gslab_python/issues/23 is created, then this will take care of the issue (see my recent https://github.com/gslab-econ/gslab_python/issues/23#issuecomment-261677763). For now, it's probably best to just use `figure.eps` as the source so everything works and then once https://github.com/gslab-econ/gslab_python/issues/23 is done, we can update.

I

**Comment by lboxell** [comment]
@yuchuan2016,

We should also add an example of 'source/input' from https://github.com/gslab-econ/template/issues/19#issuecomment-262024738.

**Comment by yuchuan2016** [comment]
On hold.

**Comment by yuchuan2016** [comment]
On hold.

**Comment by yuchuan2016** [comment]
Hi @lboxell ,

I want to wrap up this task. I'm somewhat lost about the status of other branches in this repo, since I have not been working on `gslab_python` for quite a long time. Do you think it's still useful to merge my changes into the master? Perhaps it may create conflicts with other branches in the repo. All my changes are in [this comment](https://github.com/gslab-econ/template/issues/20#issuecomment-260723272). If you think it's useful, I will create a pull request.

Thanks a lot!

**Comment by lboxell** [comment]
@yuchuan2016,

Yup, I think we can still continue as planned.

**Comment by yuchuan2016** [comment]
Merged #26.

Closing this task. Major changes are summarized in [this comment](https://github.com/gslab-econ/template/issues/20#issuecomment-260723272).
